Audiobook Player/Player/Containers/PlayerControlsViewController.swift

```diff
@@ -8,69 +8,225 @@
         didSet {
-            self.playPauseButton.setImage(self.isPlaying ? self.pauseImage : self.playImage, for: UIControlState())
+            guard let colors = self.colors else {
+                return
+            }
+
+            self.rewindIcon.tintColor = colors[3]
+            self.forwardIcon.tintColor = colors[3]
+
+            self.artwork.layer.shadowOpacity = 0.2 + Float(1.0 - colors[0].luminance) * 0.2
         }
     }
 
-    let playImage = UIImage(named: "playButton")
-    let pauseImage = UIImage(named: "pauseButton")
-
     override func viewDidLoad() {
         super.viewDidLoad()
 
+        self.originalHeight = self.artworkHeight.constant
         self.isPlaying = PlayerManager.sharedInstance.isPlaying
 
-        self.artwork.layer.shadowColor = UIColor.flatBlack().cgColor
-        self.artwork.layer.shadowOffset = CGSize(width: 0, height: 4)
-        self.artwork.layer.shadowOpacity = 0.6
-        self.artwork.layer.shadowRadius = 6.0
+        self.artwork.layer.shadowColor = UIColor.black.cgColor
+        self.artwork.layer.shadowOffset = CGSize(width: 0.0, height: 4.0)
+        self.artwork.layer.shadowOpacity = 0.2
+        self.artwork.layer.shadowRadius = 12.0
         self.artwork.clipsToBounds = false
 
+        self.rewindIcon.title = "−30s"
+        self.forwardIcon.title = "+30s"
+
+        self.rewindIcon.alpha = self.jumpIconAlpha
+        self.forwardIcon.alpha = self.jumpIconAlpha
+
         NotificationCenter.default.addObserver(self, selector: #selector(self.onBookPlay), name: Notification.Name.AudiobookPlayer.bookPlayed, object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(self.onBookPause), name: Notification.Name.AudiobookPlayer.bookPaused, object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(self.onBookPause), name: Notification.Name.AudiobookPlayer.bookEnd, object: nil)
+
+        self.setupGestures()
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
     }
 
-    // skip time forward
-    @IBAction func forward(_ sender: Any) {
-        PlayerManager.sharedInstance.forward()
-    }
-
-    // skip time backwards
-    @IBAction func rewind(_ sender: Any) {
-        PlayerManager.sharedInstance.rewind()
-    }
-
     // toggle play/pause of book
-    @IBAction func play(_ sender: Any) {
+    @IBAction private func play(_ sender: Any) {
         PlayerManager.sharedInstance.playPause()
 
         self.isPlaying = PlayerManager.sharedInstance.isPlaying
     }
 
-    @objc func onBookPlay() {
+    @objc private func onBookPlay() {
         self.isPlaying = true
     }
 
-    @objc func onBookPause() {
+    @objc private func onBookPause() {
         self.isPlaying = false
     }
+
+    // MARK: Gesture recognizers
+
+    private func setupGestures() {
+        self.pan = UIPanGestureRecognizer(target: self, action: #selector(handlePan))
+        self.pan!.delegate = self
```

Comment:
is `!` necessary?

Reply:
Nope, I think this is left from earlier code.